### PR TITLE
States API Documentation and Cleanup

### PR DESCRIPTION
**Pull request by bgfoshay**
## What’s changing?

Adding/Improving documentation for States API

## Why?

NAC-932, NAC-934, NAC-935, NAC-945. Adding documentation for recently introduced States API

## This PR has:

- [ ] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [x] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [ ] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: states/docs/openapi/get_states.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    - "States"
+  summary: "Retrieve metadata for all States"
+  description: "Retrieves metadata records for every State in the Collaboration database"
+  responses:
+    '200':
+      $ref: './schemas/states.yaml#/components/responses/200_Get_States'
+    '401':
+      $ref: './schemas/states.yaml#/components/responses/401'
+    '429':
+      $ref: './schemas/states.yaml#/components/responses/429'
```
Comment (kingcomma) [inline states/docs/openapi/get_states.yaml:13]: *Nitpick (non-blocking):* Could probably drop this 429 response for now. Some form of rate-limiting on the internal API may not be a bad idea in the future, but I assume it's not yet implemented?

File: states/docs/openapi/schemas/state-record.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    email: "va_agency_worker@gmail.com"
+    phone: "7031234567"
+    region: "Region1"
+  MD:
+    id: 2
+    state: "Maryland"
+    state_abbreviation: "MD"
+    email: "md_agency_worker@gmail.com"
+    phone: "4101234567"
+    region: "Region1"
```
Comment (kingcomma) [inline states/docs/openapi/schemas/state-record.yaml:44]: *Nitpicks:* A few small things —
* Unless there's a reason not to, let's stick to using the EA, EB, EC example states
* I opened up a ticket for this more generally across the repo, but we should stick to TLDs that are reserved for testing purposes for placeholder emails like this. E.g., `no-reply@ea.example` or something similar.
* Similar to emails, we should probably avoid using potentially real phone numbers. I think numbers in the `555-01xx` range are considered safe (https://en.wikipedia.org/wiki/555_(telephone_number))

File: states/docs/openapi/index.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Match Resolution API base spec
```
Comment (kingcomma) [inline states/docs/openapi/index.yaml:1]:
```suggestion
# State Metadata API base spec
```

**Review by kingcomma — COMMENTED** [review]
Looks great overall — I pulled it up using the [OpenAPI preview extension in VS Code](https://marketplace.visualstudio.com/items?itemName=42Crunch.vscode-openapi) and formatting/structure checks out. Just a few small nitpicks, mostly around example data, if you wouldn't mind. 

Really appreciate this getting picked up 👍

File: states/docs/openapi/get_states.yaml (hunk 1 above)
Comment (bgfoshay) [reply states/docs/openapi/get_states.yaml:13]: Yes, you're right that we can drop this for now. I removed it.

File: states/docs/openapi/schemas/state-record.yaml (hunk 2 above)
Comment (bgfoshay) [reply states/docs/openapi/schemas/state-record.yaml:44]: 👍These all make sense. I'll adjust the examples.

File: states/docs/openapi/index.yaml (hunk 3 above)
Comment (bgfoshay) [reply states/docs/openapi/index.yaml:1]: Whoops. Good catch on this one.

**Review by kingcomma — APPROVED** [review]
👍

**Review by ccaprio97 — APPROVED** [review]
LGTM

**Pull request merged**